core/amber/src/main/python/core/models/operator.py

```diff
@@ -96,6 +97,60 @@ def on_finish(self, port: int) -> Iterator[Optional[TupleLike]]:
         yield
 
 
+class BatchOperator(TupleOperatorV2):
+    """
+    Base class for batch-oriented operators. A concrete implementation must
+    be provided upon using.
+    """
+    BATCH_SIZE: Optional[int] = None  # must be a positive integer
+    DEFAULT_VALUE: int = 50 # if BATCH_SIZE is
+
+    def __init__(self):
+        super().__init__()
+        self.__internal_is_source: bool = False
+        self.__batch_data: Mapping[int, List[Tuple]] = defaultdict(list)
+        # must be a positive integer
+        # assert self.BATCH_SIZE is None or (isinstance(self.BATCH_SIZE, int) and self.BATCH_SIZE > 0)
+
+    @overrides.final
+    def process_tuple(self, tuple_: Tuple, port: int) -> Iterator[Optional[TupleLike]]:
+        if self.BATCH_SIZE is None or not isinstance(self.BATCH_SIZE, int) or self.BATCH_SIZE <= 0:
```

Comment:
I suggest define a `_valid_batch_size` method to do the check. And this check should be added multiple places:
1. when batch operator is initiated
2. when batch size is explicitly altered. 

thus, I suggest to add such check to `__set_attr__` method.  We do not need to have this check before processing each tuple.

Reply:
fix